crates/holochain/src/core/ribosome/guest_callback/post_commit.rs

```diff
@@ -87,6 +95,45 @@ impl From<Vec<PostCommitCallbackResult>> for PostCommitResult {
     }
 }
 
+pub async fn spawn_post_commit<Ribosome: 'static>(
+    ribosome: Ribosome,
+    workspace: HostFnWorkspace,
+    network: HolochainP2pCell,
+    keystore: KeystoreSender,
+    zomed_headers: Vec<(Option<Zome>, SignedHeaderHashed)>,
+) where
+    Ribosome: Clone + RibosomeT + Send,
+{
+    for (maybe_zome, grouped_headers) in &zomed_headers
+        .into_iter()
+        .group_by(|(zome, _shh)| zome.clone())
+    {
+        if let Some(zome) = maybe_zome {
+            let headers = grouped_headers
+                .into_iter()
+                .map(|(_zome, shh)| shh)
+                .collect::<Vec<_>>();
+            let zome = zome.clone();
+            let ribosome = ribosome.clone();
+            let workspace = workspace.clone();
+            let network = network.clone();
+            let keystore = keystore.clone();
+            tokio::task::spawn(async move {
```

Comment:
I don't think it's a great idea to spawn an unbounded amount of tokio tasks here. We could either have a Semaphore to limit the amount of tasks or better yet have a single spawned background task that could concurrently work through the calls..